### DISCUSSION TITLE
While Performing the SDC Setup for a State, we are facing keycloak and ingestion-ms build Error.

### DISCUSSION THREAD

**Discussion by lokeshshivamurthy** [Sunbird-cQube/community#867] [Q&A]
Repository :  https://github.com/Sunbird-cQube/cqube-devops.git
Ubuntu Version: 22.04
Branch: release-v5.0.3

Issue: When we are performing SDC Setup for a State, during the Installation Process we are facing keycloak and ingestion-ms build error.

a. Keycloak Error:
    Error Building keycloak_ms, pull access denied for jboss/keycloak. Basically during installation, joss/keycloak:15.0.2 image is not being 
    downloaded. As a result, installation is getting halted. (Screenshot Attached)
![Keycloak_Error_1](https://github.com/Sunbird-cQube/community/assets/134672230/759476ef-6363-4193-9845-13591648c1b6)

![Keycloak_Error_2](https://github.com/Sunbird-cQube/community/assets/134672230/cc2857b5-184a-487c-8e8d-4faf0d0a2063)

Steps taken to resolve this Issue:
 1. The joss/keycloak:15.0.2 image which was available with us, we loaded it into our VM and continued with Installation. Keycloak error got passed, but we are also facing ingestion_ms image error.
 
b. Ingestion_ms Error:
   We are getting "Pull access denied for ingestion_ms and Basically it is not able to download the ingestion_ms docker image during 
   installation setup.
![Ingestion_ms Error](https://github.com/Sunbird-cQube/community/assets/134672230/3eb3ca49-bcf3-4ed4-b8ba-98957224fdf3)

Can these two issued be resolved, such that we can pull the latest code base from Sunbird Repo.

**Comment by PavitraIR** [comment]
@lokeshshivamurthy 

Thanks for reaching out, Let me check from end then I'll get back to you

**Comment by PavitraIR** [comment]
@lokeshshivamurthy 

We can able to build the ingestion_ms docker, can you please retest it and let me know. Could you please tell me which storage type you are installing

**Reply by lokeshshivamurthy** [reply]
What about keycloak-ms  ? Are you able to build keycloak  ?

**Reply by PavitraIR** [reply]
JBoss Keycloak has been officially removed as an alternative method, and we are now concentrating on integrating with the most recent Keycloak image. We are currently investigating this change, and I will get back to you on this.

**Reply by PavitraIR** [reply]
@lokeshshivamurthy 

We have gone through the issue and found that the docker hub has officially discontinued jboss keycloak images.We have to use an alternative image which is **quay keycloak.**
We need to explore and work on it to make this work on **/auth** using which we can avoid a lot of developmental changes

**Reply by PavitraIR** [reply]
Hi @lokeshshivamurthy 

I have tested end-to-end using the keycloak latest image, and the Keycloak issue has been successfully resolved. Both the Keycloak UI and the overall UI are functioning as expected. The ingestion process is working fine.
